src/jit/codegencommon.cpp

```diff
@@ -667,6 +665,9 @@ regMaskTP Compiler::compNoGCHelperCallKillSet(CorInfoHelpFunc helper)
         case CORINFO_HELP_ASSIGN_REF:
         case CORINFO_HELP_CHECKED_ASSIGN_REF:
             return RBM_CALLEE_GCTRASH_WRITEBARRIER;
+        case CORINFO_HELP_PROF_FCN_LEAVE:
+            // In case of Leave profiler callback, we need to preserve liveness of REG_PROFILER_RET_SCRATCH on ARMARCH.
+            return RBM_CALLEE_TRASH_NOGC & ~RBM_PROFILER_RET_SCRATCH;
```

Comment:
I think this construction is confusing. I think you should define a new:
```
RBM_CALLEE_GCTRASH_PROF_FCN_LEAVE
```
set for each architecture (or just arm32/arm64), just as we have similar things for other cases. Then, in target.h, where this would be defined, you can have a nice explicit comment about the ABI for these helpers (like we now have for `CORINFO_HELP_ASSIGN_REF`, for example).